src/Security/MemberAuthenticator/Authenticator.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+namespace SilverStripe\Security\MemberAuthenticator;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\Session;
+use SilverStripe\Dev\Debug;
+use SilverStripe\ORM\ValidationResult;
+use InvalidArgumentException;
+use SilverStripe\Security\Authenticator as BaseAuthenticator;
+use SilverStripe\Security\Security;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\LoginAttempt;
+
+/**
+ * Authenticator for the default "member" method
+ *
+ * @author Sam Minnee <sam@silverstripe.com>
+ * @author Simon Erkelens <simonerkelens@silverstripe.com>
+ */
+class Authenticator implements BaseAuthenticator
+{
+
+    public function supportedServices()
+    {
+        // Bitwise-OR of all the supported services in this Authenticator, to make a bitmask
+        return BaseAuthenticator::LOGIN | BaseAuthenticator::LOGOUT | BaseAuthenticator::CHANGE_PASSWORD
+            | BaseAuthenticator::RESET_PASSWORD;
+    }
+
+    /**
+     * @param array $data
+     * @param null|ValidationResult $result
+     * @return null|Member
+     */
+    public function authenticate($data, &$result = null)
+    {
+        $result = null;
+
+        // Find authenticated member
+        $member = $this->authenticateMember($data, $result);
+
+        // Optionally record every login attempt as a {@link LoginAttempt} object
+        $this->recordLoginAttempt($data, $member, $result->isValid());
+
+        if ($member) {
+            Session::clear('BackURL');
+        }
+
+        return $result->isValid() ? $member : null;
+    }
+
+    /**
+     * Attempt to find and authenticate member if possible from the given data
+     *
+     * @param array $data Form submitted data
+     * @param ValidationResult $result
```

Comment:
Come to think of it, would the API be more concise if we change the $message argument to a ValidationResult instance? That way we wouldn't get pre-emptive serialisation of messages, code could more easily throw / return the result.

It could ALSO replace the $success, since you could query $result->isValid(). :D

We would need to make this change all the way down the stack (Authenticator interface, etc).

What do you think about this @Firesphere ?
++++
Okay, Agreed, took me less work than initially expected ;)